os/arch/arm/src/common/up_createstack.c

```diff
@@ -185,7 +189,12 @@ int up_create_stack(FAR struct tcb_s *tcb, size_t stack_size, uint8_t ttype)
 #endif
 		{
 			/* Use the user-space allocator if this is a task or pthread */
+#ifdef CONFIG_MPU_STACK_OVERFLOW_PROTECTION	/* address must be aligned to 32 byte for MPU region */
+			stack_size = stack_size + CONFIG_MPU_STACK_GUARD_SIZE;
+			tcb->stack_alloc_ptr = (uint32_t *)kumm_memalign(MPU_REGION_ADDRESS_ALIGNMENT, stack_size);
```

Comment:
As my understanding, guard should be aligned to its size. But you made alignment with guard + stack size. Is it okay?

Reply:
yes, it if fine, we are aligning the address assigned for the total size including guard memory.
Guard address will be the stack_alloc_ptr address which will be aligned to guard size/mpu region size.